mp4parse/src/lib.rs

```diff
@@ -1173,129 +1173,141 @@ fn read_flac_metadata<T: Read>(src: &mut BMFFBox<T>) -> Result<FLACMetadataBlock
     })
 }
 
-fn read_esds<T: Read>(src: &mut BMFFBox<T>) -> Result<ES_Descriptor> {
+fn find_descriptor<'a>(data: &'a [u8], esds: &mut ES_Descriptor) -> Result<()> {
```

Comment:
It's only the binding that's mutable; it allows `remains` to be reassigned in the loop.  `data` isn't becoming mutable; that'd be something like `let remains: &mut [u8] = data;`, which would be a compile error.